### ISSUE TITLE
How to change column-width?

### ISSUE THREAD

**Issue by LaurinKerkloh** [westonganger/rodf#39]
Sorry if the question is dumb, but i don't understand how to do this.

@cases = get_searchresults
    ss = RODF::Spreadsheet.new
    ss.style 'width', family: :cell do
      property :column, 'column-width' => '4.0cm'
    end
    t = ss.table 'Fälle'
    r = t.row
    r.cell 'Fallnummer', style: 'width'
    r.cell 'Mitgliedsnummer', style: 'width'
    r.cell 'empfangen'
    r.cell 'Status'
    @cases.each do |c|
      r = t.row
      r.cell c.casenumber
      r.cell c.membernumber
      r.cell c.received.strftime("%Y-%m-%d %H:%M:%S")
      r.cell c.state
    end
    ss.write_to 'storage/searchexport.ods'
    redirect_to cases_search_path

Thanks for your help!

**Comment by westonganger** [comment]
I dont think that I was able to make this work within [`spreadsheet_architect`](https://github.com/westonganger/spreadsheet_architect) either. 

Would love a PR that fixes this functionality.

**Comment by dmolesUC** [comment]
I don't have time to do a PR right now, but I was able to get it to work with something like:

```ruby
RODF::Spreadsheet.new do
  style('col-default', family: :column) do
    property(:column, 'column-width' => '1.234in')
  end

  ss_table = table('Sheet 1') do
    # build table
  end

  # by default table() doesn't seem to create any columns,
  # and there's no DSL methods for them, but we can still
  # create and add them by hand
  col_count.times do   # I assume we know how many columns we have
    ss_col = RODF::Column.new(style: 'col-default')
    ss_table.columns << ss_col
  end
end
```

**Comment by westonganger** [comment]
@dmolesUC would it make sense for RODF to automatically create the columns upon calling of either `cell`, `add_rows`, or `add_cells` ?

**Comment by dmolesUC** [comment]
I don't think it's necessary unless you actually do want to add column properties — spreadsheets seem to work fine without them. I think a `column` DSL method (similar to `row`) would be more consistent with the rest of the API. I'm not really familiar with `Builder::XmlMarkup`, though, so I don't know how much trouble that would be or what would work best.

**Comment by westonganger** [comment]
Actually it turns out that there already is a `column` method available [spec/table_spec.rb#L34](https://github.com/westonganger/rodf/blob/master/spec/table_spec.rb#L34). I have now documented this in the Readme. 

I have also added a method to Table called `set_column_width(widths_array)`. See the Readme for further usage.

It was difficult to understand the existing code because of a metaprogramming technique that was used, the `Container.contains` method (`contains :foobar`) used `define_method`. I have [re-written the RODF code](https://github.com/westonganger/rodf/commit/b19a4b5684368f9dd7ee16d58de56d8cb505dc58) so that the methods aren't metaprogrammed anymore and now the code makes much more sense.
